### DISCUSSION TITLE
Corepack does not work with Yarn 4.0.2

### DISCUSSION THREAD

**Discussion by GauBen** [vercel/community#5122] [Bug Report]
### Page to Investigate

https://vercel.com/docs/deployments/configure-a-build#corepack

### Steps to Reproduce

1. Set `ENABLE_EXPERIMENTAL_COREPACK` to 1 in build settings
2. Set `"packageManager": "yarn@4.0.2"` in package.json
3. Set `yarn install` as install command

Build will fail with the following logs:

```
Vercel CLI 32.6.1
Running "install" command: `yarn install`...
file:///vercel/path0/.vercel/cache/corepack/home/yarn/4.0.2/yarn.js:4
  <minified code omitted>
Error: Dynamic require of "util" is not supported
    at file:///vercel/path0/.vercel/cache/corepack/home/yarn/4.0.2/yarn.js:4:394
    at file:///vercel/path0/.vercel/cache/corepack/home/yarn/4.0.2/yarn.js:4:3572
    at file:///vercel/path0/.vercel/cache/corepack/home/yarn/4.0.2/yarn.js:4:478
    at file:///vercel/path0/.vercel/cache/corepack/home/yarn/4.0.2/yarn.js:9:45688
    at file:///vercel/path0/.vercel/cache/corepack/home/yarn/4.0.2/yarn.js:4:478
    at file:///vercel/path0/.vercel/cache/corepack/home/yarn/4.0.2/yarn.js:392:1509
    at file:///vercel/path0/.vercel/cache/corepack/home/yarn/4.0.2/yarn.js:734:11903
    at ModuleJob.run (node:internal/modules/esm/module_job:192:25)
    at async DefaultModuleLoader.import (node:internal/modules/esm/loader:228:24)
    at async loadESM (node:internal/process/esm_loader:40:7)
Node.js v20.5.1
Error: Command "yarn install" exited with 1
```

**Comment by DungGramer** [comment]
1. Execute `yarn install` locally and fix any encountered bugs.
2. Perform a local build and resolve any arising issues.
3. Finally, deploy locally using `vercel` ([vercel CLI docs](https://vercel.com/docs/cli)).

**Comment by bd-charu** [comment]
Have you resolved this? I deployed 8 hours ago and I am now getting this error. I have set it temporarily to use yarn v1 for it to "work"

**Reply by GauBen** [reply]
I gave up using corepack, I just use yarn 4 with global yarn 1 setup

**Comment by cermakjiri** [comment]
As a hotfix, changing `"type": "module"` to `"type": "commonjs"` in root `package.json` has fixed it for me. (`vercel@npm:33.7.0`)
